router.go

```diff
@@ -375,6 +383,55 @@ func (app *App) register(methods []string, pathRaw string, group *Group, handler
 	}
 }
 
+func (app *App) routeExists(method string, pathRaw string) bool {
+	return slices.ContainsFunc(app.stack[app.methodInt(method)], func(r *Route) bool {
+		return r.path == pathRaw
+	})
+}
+
+// RemoveRoute is used to remove a route from the stack
+// You should call RebuildTree after using this to ensure consistency of the tree
+func (app *App) RemoveRoute(path string, middlewareCount int, methods ...string) {
```

Comment:
That `middlewareCount` param is very confusing. What's the purpose of it?

Reply:
I agree, I do not like having to pass in the middlewareCount however.  The reason I believe this is needed is because the register method currently uses the length of handlers to increment the global routeCount.  

Here we can see that the handlers count should always be at least one. 
```go
func (app *App) register(methods []string, pathRaw string, group *Group, handler Handler, middleware ...Handler) {
	handlers := middleware
	if handler != nil {
		handlers = append(handlers, handler)
	}
```
^
|  more register method code between
v
```go
	// Increment global handler count
	atomic.AddUint32(&app.handlersCount, uint32(len(handlers))) //nolint:gosec // Not a concern
```

When removing a route we need to ensure that the app.handersCount is updated as well.  Since the user can pass in optional middleware this count can be greater than one for a given route.

```go 
func (app *App) deleteRoute(path string, methods []string, middlewareCount int) {
	app.mutex.Lock()
	defer app.mutex.Unlock()

	// No middleware passed in, increment to account for routes handler function
	if middlewareCount == 0 {
		middlewareCount++
	}

	//Decrement global handler count
	atomic.AddUint32(&app.handlersCount, ^uint32(middlewareCount-1)) //nolint:gosec // Not a concern
```

I have also thought of just passing in the registered middleware to RemoveRoute method and then passing the number of handlers into the deleteRoute method.  Not sure if that would be any less confusing.

I am open to suggestions.

Reply:
Actually now that I am thinking about this more I think I should be able to get this from the existing route with that path and method.   I am starting to work on the RemoveRouteByName method that @efectn asked for so I will have to figure out how to get the counts from the existing routes since I will have to find all routes that contain the name of the route that needs to be deleted anyway.

Reply:
I figured something better out.  Working on cleaning it up now.

Reply:
@gaby Just cleaned up the call to deleteRoute in my latest commit.  I was able to do away with passing in any additional params to the method I solved it by doing this instead within the deleteRoute method.

```go
		// Find the index of the route to remove
		index := slices.IndexFunc(app.stack[m], matchFunc)
		if index == -1 {
			continue // Route not found
		}

		route := app.stack[m][index]

		// Decrement global handler count
		atomic.AddUint32(&app.handlersCount, ^uint32(len(route.Handlers)-1)) // nolint:gosec // Not a concern
		// Decrement global route position
		atomic.AddUint32(&app.routesCount, ^uint32(0))

```